### DISCUSSION TITLE
relations with combined queries

### DISCUSSION THREAD

**Discussion by CrisGuzmanS** [laravel/framework#46330] [Q&A]
Hello! ✋ i am not sure if this is the correct place to put this problem, i want to tell you about it to know if you have an approach.

I have this 3 models:
* Person
* Organization
* Publication

The Organization has a many to many relation with Publication (through organization__publication table).
The Person has a many to many relation with Publication (through person__publication table).
The Organization has many to many relation with Person (through organization__person table).

In my Organization model i have my relation between Organization and Publication:
```php
public function publications()
    {
        return $this->belongsToMany(
            Publication::class,
            'organization__publication',
            'organization_id',
            'publication_id'
        );
```

I want to get deep publications of the organization (the organization publications and the person publication that belongs to the organization). This is why i implemented this method:
```php
public function deepPublications()
    {
        return Publication::whereHas('persons', function ($query) {
            $query->whereOrganization($this);
        })->orWhere(function ($query) {
            $query->whereOrganization($this);
        });
    }
```

as you see i achieved what i want. The problem is when i want to do this:
```php
Organization::whereHas('deepPublications', function($query){
    //...
});

// and

$organization->deepPublications()->count();

// and

$organization->where('deepPublications','>',3);

```

as you can see in my deepPublications method i create a new query, i was thinking to create an scope method (scopeDeepPublications), but i am still not achieving what i want. Do you know any approach to do this?

**Comment by parallels999** [comment]
Did you try something like this??
```php
Organization::whereHas('publications.persons', function($query){
    //...
});
```
